Wrote editlab view and html

BMHebert opened the pull request:
Implemented a view and html page for viewing and editing labs that a supervisor can access. Also miscellaneous tiny changes.

BMHebert committed:
Wrote editlab view and html

Implemented a view and html page for viewing and editing labs that a supervisor can access. Also miscellaneous tiny changes.

BMHebert committed:
Merge branch 'main' into BMHebert2

BMHebert committed:
Update edituser.html

Tiny bug fix in edituser.html

BMHebert committed:
Update CourseManagement.py

Small change to CourseManagement.inputErrorChecker since it wasn't really written with much care in the first case.

BMHebert commented:
Full list of changes since I was somewhat vague before:

- Heavily editlab.html to display database information and implemented edit functionality.
- Wrote view for editlab with functionality to process input from editlab.html
- Fixed some small formatting issues in edituser.html
- Added __str__ method for UserProfile
- Fixed LabManagement.editLab since it wasn't properly saving the object to the database.
- Improved CourseManagement.inputErrorChecker since it wasn't really written with much care in the first case.

BMHebert committed:
Merge branch 'main' into BMHebert2

WajahatAliShah1 reviewed (approved):
All changes look good, once second review is in, we can merge and update our tests to test functionality of saving new or edited information to database.

BMHebert committed:
Updated the model for lab

Updated the models for lab to include a course foreignkey and removed labs manytomany field from course. Also removed instructor foreignkey from lab since it is no longer necessary. Got the rest of the code working with the changes to model as well.

BMHebert committed:
Update edituser.html

Slight formatting change

BMHebert committed:
Merge branch 'main' into BMHebert2
